components/renku_data_services/authz/authz.py

```diff
@@ -831,6 +851,19 @@ async def upsert_project_members(
                 # The existing relationships should be deleted if all goes well and added back in if we have to undo
                 existing_rel = existing_rels[0]
                 if existing_rel.relationship != rel:
+                    if existing_rel.relationship.relation == _Relation.owner.value:
+                        if existing_owners_rels is None:
+                            existing_owners_rels = await self._get_resource_owners(
+                                resource_type, resource_id, consistency
+                            )
+                        if len(existing_owners_rels) == 1:
```

Comment:
I think there's still an edge case here if there's two or more owners and all of them get unset in the same request. This code wouldn't fail but you'd still end up with no owners.